crates/sui-graphql-rpc/src/types/event/lookups.rs

```diff
@@ -0,0 +1,197 @@
+    ));
+
+    if let Some(sender) = sender {
+        return query.filter(format!("sender = {}", bytea_literal(sender.as_slice())));
+    }
+
+    query
+}
+
+pub(crate) fn select_sender(sender: SuiAddress) -> RawQuery {
+    select_ev(Some(sender), "event_senders")
+}
+
+pub(crate) fn select_event_type(event_type: &TypeFilter, sender: Option<SuiAddress>) -> RawQuery {
+    match event_type {
+        TypeFilter::ByModule(ModuleFilter::ByPackage(p)) => {
+            filter!(
+                select_ev(sender, "event_struct_package"),
+                format!("package = {}", bytea_literal(p.as_slice()))
+            )
+        }
+        TypeFilter::ByModule(ModuleFilter::ByModule(p, m)) => {
+            filter!(
+                select_ev(sender, "event_struct_module"),
+                format!(
+                    "package = {} and module = {{}}",
+                    bytea_literal(p.as_slice())
+                ),
+                m
+            )
+        }
+        TypeFilter::ByType(tag) => {
+            let p = tag.address.to_vec();
+            let m = tag.module.to_string();
+            let exact = tag.to_canonical_string(/* with_prefix */ true);
+            let t = exact.split("::").nth(2).unwrap();
+
+            let (table, col_name) = if tag.type_params.is_empty() {
+                ("event_struct_name", "type_name")
+            } else {
+                ("event_struct_name_instantiation", "type_instantiation")
+            };
+
+            filter!(
+                select_ev(sender, table),
+                format!(
+                    "package = {} and module = {{}} and {} = {{}}",
+                    bytea_literal(p.as_slice()),
+                    col_name
+                ),
+                m,
+                t
+            )
+        }
+    }
+}
+
+pub(crate) fn select_emit_module(
+    emit_module: &ModuleFilter,
+    sender: Option<SuiAddress>,
+) -> RawQuery {
+    match emit_module {
+        ModuleFilter::ByPackage(p) => {
+            filter!(
+                select_ev(sender, "event_emit_package"),
+                format!("package = {}", bytea_literal(p.as_slice()))
+            )
+        }
+        ModuleFilter::ByModule(p, m) => {
+            filter!(
+                select_ev(sender, "event_emit_module"),
+                format!(
+                    "package = {} and module = {{}}",
+                    bytea_literal(p.as_slice())
+                ),
+                m
+            )
+        }
+    }
+}
+
+/// Given a `RawQuery` representing a query for events, adds ctes and corresponding filters to
+/// constrain the query. If a transaction digest is specified, we add a `tx_eq` CTE from which the
+/// lower and upper bounds will draw from. This also means an additional check that the `tx_eq` CTE
+/// returns a non-empty result. Otherwise, the lower bound is determined from `page.after()` or 0 by
+/// default. The upper bound is determined by `page.before()`, or the current latest tx sequence
```

Comment:
```suggestion
/// default. The upper bound is determined by `page.before()`, or the current latest transaction sequence
```
Or `tx_sequence_number` and delete next line